### ISSUE TITLE
TimeSeries panel: Treshold point connection not working properly

### ISSUE THREAD

**Issue by Knocks83** [grafana/grafana#34166]
**What happened**:
Time Series graph with 2 queries, I tried to enable the connect null values "treshold" mode but depending on the time range it fills differently

As you can see, even if the points are distant about 3 minutes one from the other, with a treshold of <1h it doesn't connect them:
![image](https://user-images.githubusercontent.com/22529815/118409659-d67a9800-b68b-11eb-8c64-df634a24f6d6.png)

Even by filtering for the last 6 hours doesn't connect them, but if I filter it for the last 2 days some data are connected (I disabled the dots because they were too much):
![image](https://user-images.githubusercontent.com/22529815/118409734-340ee480-b68c-11eb-9f90-ae8d7c086735.png)

I noticed that in the last part they are connected, so I tried focusing on that time period and that's what I got:
![image](https://user-images.githubusercontent.com/22529815/118409806-89e38c80-b68c-11eb-879a-567ee6e0c145.png)

As you can see, they are connected

**What you expected to happen**:
It should connect a point to the next one only if the time difference is less than 1h

**How to reproduce it (as minimally and precisely as possible)**:
In my case I'm using a 2-query graph, but I don't know if this bug happens also with single-query graphs.
Create a 2-query Time series graph, enable connect null values.
This is the JSON of my panel:
```
{
  "id": 23763571993,
  "gridPos": {
    "h": 19,
    "w": 24,
    "x": 0,
    "y": 0
  },
  "type": "timeseries",
  "title": "Environment",
  "datasource": "Localhost",
  "pluginVersion": "8.0.0-beta1",
  "fieldConfig": {
    "defaults": {
      "custom": {
        "drawStyle": "line",
        "lineInterpolation": "smooth",
        "barAlignment": 0,
        "lineWidth": 1,
        "fillOpacity": 10,
        "gradientMode": "none",
        "spanNulls": 3600000,
        "showPoints": "always",
        "pointSize": 5,
        "stacking": {
          "mode": "none",
          "group": "A"
        },
        "axisPlacement": "auto",
        "axisLabel": "Temperature",
        "scaleDistribution": {
          "type": "linear"
        },
        "hideFrom": {
          "tooltip": false,
          "graph": false,
          "legend": false
        },
        "thresholdsStyle": {
          "mode": "off"
        },
        "lineStyle": {
          "fill": "solid"
        }
      },
      "color": {
        "mode": "palette-classic"
      },
      "thresholds": {
        "mode": "absolute",
        "steps": [
          {
            "color": "green",
            "value": null
          }
        ]
      },
      "mappings": [],
      "decimals": 1,
      "max": 40,
      "min": 10,
      "unit": "celsius"
    },
    "overrides": [
      {
        "matcher": {
          "id": "byName",
          "options": "Humidity"
        },
        "properties": [
          {
            "id": "color",
            "value": {
              "fixedColor": "blue",
              "mode": "fixed"
            }
          }
        ]
      },
      {
        "matcher": {
          "id": "byName",
          "options": "Temperature"
        },
        "properties": [
          {
            "id": "color",
            "value": {
              "fixedColor": "red",
              "mode": "fixed"
            }
          }
        ]
      },
      {
        "matcher": {
          "id": "byName",
          "options": "Humidity"
        },
        "properties": [
          {
            "id": "unit",
            "value": "percent"
          },
          {
            "id": "max",
            "value": 100
          },
          {
            "id": "custom.axisLabel",
            "value": "Humidity"
          }
        ]
      }
    ]
  },
  "options": {
    "tooltip": {
      "mode": "single"
    },
    "legend": {
      "displayMode": "list",
      "placement": "bottom",
      "calcs": [
        "lastNotNull",
        "max",
        "min"
      ]
    }
  },
  "targets": [
    {
      "alias": "Temperature",
      "groupBy": [
        {
          "params": [
            "$__interval"
          ],
          "type": "time"
        },
        {
          "params": [
            "null"
          ],
          "type": "fill"
        }
      ],
      "hide": false,
      "measurement": "environment",
      "orderByTime": "ASC",
      "policy": "default",
      "query": "SELECT mean(\"temperature\") FROM \"environment\" WHERE $timeFilter GROUP BY time($__interval) fill(null)",
      "rawQuery": false,
      "refId": "A",
      "resultFormat": "time_series",
      "select": [
        [
          {
            "params": [
              "temperature"
            ],
            "type": "field"
          },
          {
            "params": [],
            "type": "mean"
          }
        ]
      ],
      "tags": []
    },
    {
      "alias": "Humidity",
      "groupBy": [
        {
          "params": [
            "$__interval"
          ],
          "type": "time"
        },
        {
          "params": [
            "null"
          ],
          "type": "fill"
        }
      ],
      "measurement": "environment",
      "orderByTime": "ASC",
      "policy": "default",
      "query": "SELECT mean(\"value\") FROM \"Humidity\" WHERE $timeFilter GROUP BY time($__interval) fill(null)",
      "rawQuery": false,
      "refId": "B",
      "resultFormat": "time_series",
      "select": [
        [
          {
            "params": [
              "humidity"
            ],
            "type": "field"
          },
          {
            "params": [],
            "type": "mean"
          }
        ]
      ],
      "tags": []
    }
  ],
  "transparent": true,
  "timeFrom": null,
  "timeShift": null
}
```

**Environment**:
- Grafana version: 8.0.0 beta 1
- Data source type & version: InfluxDB 1.8.5
- OS Grafana is installed on: Ubuntu 20.04
- User OS & Browser: Windows 10 - Firefox

**Comment by Knocks83** [comment]
Update:
I tried setting that graph plus the one below with the correct settings (treshold < 10m) and they behave differently:
![image](https://user-images.githubusercontent.com/22529815/118410127-36723e00-b68e-11eb-9fc1-28d35d27f05e.png)

The graph below connects when it shouldn't, and the one up doesn't connect. There are a few connections which, if you care, are points registered with respectively 40 and 20 seconds of difference.

**Comment by oscarkilhed** [comment]
Only able to reproduce this with multiple series. Works as expected with a single series.

**Comment by Knocks83** [comment]
That's my single-query panel JSON, just to be sure:
```
{
  "id": 4,
  "gridPos": {
    "h": 3,
    "w": 24,
    "x": 0,
    "y": 19
  },
  "type": "timeseries",
  "pluginVersion": "8.0.0-beta1",
  "fieldConfig": {
    "defaults": {
      "custom": {
        "drawStyle": "line",
        "lineInterpolation": "stepBefore",
        "barAlignment": 0,
        "lineWidth": 1,
        "fillOpacity": 10,
        "gradientMode": "none",
        "spanNulls": 600000,
        "showPoints": "never",
        "pointSize": 5,
        "stacking": {
          "mode": "none",
          "group": "A"
        },
        "axisPlacement": "auto",
        "axisLabel": "Status",
        "scaleDistribution": {
          "type": "linear"
        },
        "hideFrom": {
          "tooltip": false,
          "graph": false,
          "legend": false
        },
        "thresholdsStyle": {
          "mode": "off"
        }
      },
      "color": {
        "mode": "palette-classic"
      },
      "thresholds": {
        "mode": "absolute",
        "steps": [
          {
            "color": "green",
            "value": null
          }
        ]
      },
      "mappings": [],
      "decimals": 0,
      "max": 1,
      "min": 0,
      "unit": "none"
    },
    "overrides": []
  },
  "options": {
    "tooltip": {
      "mode": "single"
    },
    "legend": {
      "displayMode": "hidden",
      "placement": "bottom",
      "calcs": []
    }
  },
  "targets": [
    {
      "alias": "Status",
      "groupBy": [],
      "measurement": "environment",
      "orderByTime": "ASC",
      "policy": "default",
      "refId": "A",
      "resultFormat": "time_series",
      "select": [
        [
          {
            "params": [
              "status"
            ],
            "type": "field"
          }
        ]
      ],
      "tags": []
    }
  ],
  "transparent": true,
  "timeFrom": null,
  "timeShift": null,
  "datasource": null
}
```

**Comment by leeoniya** [comment]
@Knocks83

i believe this bug is specific to panel edit mode (please confirm). #34452 should fix this.

the test data that @oscarkilhed provided for a repro works as expected outside of panel edit.

Time | Temp
-- | --
2021-05-17 0:00:00 | 20.00
2021-05-17 0:00:01 | &nbsp;
2021-05-17 0:00:02 | 21.00
2021-05-17 0:00:03 | &nbsp;
2021-05-17 0:00:04 | &nbsp;
2021-05-17 0:00:05 | 22.00

Time | RH
-- | --
2021-05-17 0:00:01 | 10.00
2021-05-17 0:00:02 | &nbsp;
2021-05-17 0:00:03 | 5.00
2021-05-17 0:00:04 | 6.00
2021-05-17 0:00:05 | &nbsp;
2021-05-17 0:00:06 | 1.00

this is the dashboard i am testing with, using two google sheets:

```json
{
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": "-- Grafana --",
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "gnetId": null,
  "graphTooltip": 0,
  "id": 197,
  "links": [],
  "panels": [
    {
      "datasource": "Google Sheets",
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "palette-classic"
          },
          "custom": {
            "axisLabel": "",
            "axisPlacement": "auto",
            "barAlignment": 0,
            "drawStyle": "line",
            "fillOpacity": 0,
            "gradientMode": "none",
            "hideFrom": {
              "graph": false,
              "legend": false,
              "tooltip": false
            },
            "lineInterpolation": "linear",
            "lineWidth": 1,
            "pointSize": 5,
            "scaleDistribution": {
              "type": "linear"
            },
            "showPoints": "auto",
            "spanNulls": 3000,
            "stacking": {
              "group": "A",
              "mode": "none"
            },
            "thresholdsStyle": {
              "mode": "off"
            }
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 13,
        "w": 24,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "options": {
        "legend": {
          "calcs": [],
          "displayMode": "list",
          "placement": "bottom"
        },
        "tooltip": {
          "mode": "single"
        }
      },
      "targets": [
        {
          "cacheDurationSeconds": 300,
          "refId": "A",
          "spreadsheet": "1A1CmxbwIOSFU53Q8BZuKh6HzR8IFJbBxeWmuSXFkvyI"
        },
        {
          "cacheDurationSeconds": 300,
          "hide": false,
          "refId": "B",
          "spreadsheet": "1iJmWxwto_-fyKmZsNpWFNT62VjmhEyA26xboDAf87Yg"
        }
      ],
      "title": "Panel Title",
      "type": "timeseries"
    }
  ],
  "refresh": "",
  "schemaVersion": 30,
  "style": "dark",
  "tags": [],
  "templating": {
    "list": []
  },
  "time": {
    "from": "2021-05-17T04:59:30.000Z",
    "to": "2021-05-17T05:00:30.000Z"
  },
  "timepicker": {},
  "timezone": "utc",
  "title": "spanNulls-thresh-bug",
  "uid": "6wYTFnqMz",
  "version": 6
}
```

**Comment by Knocks83** [comment]
Let me add another thing: the graph on the bottom has a setting that I cannot find anymore which shows on the tooltip the nearest point to the cursor (and if you tell me where that setting is you'd do me a favor), might this be why it always connects?

**Comment by Knocks83** [comment]
@leeoniya sorry, forgot to tag you

**Comment by oscarkilhed** [comment]
@Knocks83  We'll be releasing beta-2 shortly with mentioned fix applied. If that doesn't fix the issue we'll re-open this.

**Comment by yurividal** [comment]
was this ever fixed? im having the same issue on latest 8.2.1
